chalice/chalicelib/sqs.py

```diff
@@ -0,0 +1,65 @@
+import json
+import boto3
+
+from typing import List
+from chalicelib import rand_uuid
+from chalicelib.config import MS_SQS_QUEUE_NAME, SQS_QUEUE_MSG, logger
+from chalicelib.request_handler import RequestHandler, RequestStatus
+
+
+class SqsQueueHandler:
+    """
+    AWS SQS Queue Handler for sending/receiving to/from message to
+    matrix service's SQS queue.
+    """
+
+    sqs = boto3.resource("sqs")
+
+    @staticmethod
+    def send_msg_to_ms_queue(bundle_uuids: List[str], request_id: str) -> str:
+        """
+        Send a message to matrix service's sqs queue.
+        :param bundle_uuids: UUID of the bundle.
+        :param request_id: Concatenation request id.
+        :return: Message ID of the msg being sent.
+        """
+        job_id = rand_uuid()
+
+        logger.info(f'Request ID({request_id}): Initialize the request with job id({job_id})')
+
+        RequestHandler.put_request(
+            bundle_uuids=bundle_uuids,
+            request_id=request_id,
+            job_id=job_id,
+            status=RequestStatus.INITIALIZED
+        )
```

Comment:
What if you made this a [conditional write](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.ConditionalUpdate)? That might resolve the issue of identical requests getting put into the queue.

Reply:
Thanks for pointing that out. I will use a conditional write here.